test/Riok.Mapperly.Tests/Mapping/EnumerableTest.cs

```diff
@@ -692,4 +902,16 @@ public void EnumerableMappingDisabledShouldDiagnostic()
             .Should()
             .HaveDiagnostic(new(DiagnosticDescriptors.CouldNotCreateMapping));
     }
+
+    [Fact]
+    public Task EnumerableToReadOnlyArrayPropertyShouldntForeachAdd()
+    {
+        var source = TestSourceBuilder.Mapping(
+            "A",
+            "B",
+            "class A { public ICollection<int> Value { get; } }",
+            "class B { public int[] Value { get; } }");
+
+        return TestHelper.VerifyGenerator(source);
```

Comment:
I would adjust the name of the test method to `EnumerableToReadOnlyArrayPropertyShouldDiagnostic` and add a comment (eg. `cannot create a mapping which maps to a read only array`).
Probably also change the type of assertion from snapshot to the fluent assertion style instead of the snapshot (`TestHelper.GenerateMapper(source, TestHelperOptions.AllowInfoDiagnostics) .Should()...`), since the snapshot content isn't the thing being tested here.